pkg/providers/instancetype/types.go

```diff
@@ -173,12 +173,16 @@ func ephemeralStorage(amiFamily amifamily.AMIFamily, blockDeviceMappings []*v1al
 	if len(blockDeviceMappings) != 0 {
 		switch amiFamily.(type) {
 		case *amifamily.Custom:
-			return blockDeviceMappings[len(blockDeviceMappings)-1].EBS.VolumeSize
+			// We can't know if a custom AMI is going to have a volume size.
+			volumeSize := blockDeviceMappings[len(blockDeviceMappings)-1].EBS.VolumeSize
+			if volumeSize != nil {
+				return volumeSize
+			}
 		default:
 			ephemeralBlockDevice := amiFamily.EphemeralBlockDevice()
 			for _, blockDevice := range blockDeviceMappings {
-				// If a block device mapping exists in the provider for the root volume, set the volume size specified in the provider
-				if *blockDevice.DeviceName == *ephemeralBlockDevice {
+				// If a block device mapping exists in the provider for the root volume, use the volume size specified in the provider. If not, use the default
```

Comment:
optional: Could we simplify this by doing a `lo.Find()` and then returning either the volumeSize or the default?